### PR TITLE
Add ArrayList IList tests

### PR DESCRIPTION
**Pull request by hughbe**
- Fix IList_NonGeneric tests to work with fixed size lists

/cc @ianhays @stephentoub

### CODE REVIEW

File: src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs [hunk 1]
```diff
@@ -929,14 +942,17 @@ public void ICollection_NonGeneric_Remove_EveryValue(int count)
 
         [Theory]
         [MemberData(nameof(ValidCollectionSizes))]
-        public void ICollection_NonGeneric_Remove_InvalidValue_ThrowsArgumentException(int count)
+        public void IList_NonGeneric_Remove_InvalidValue_ThrowsArgumentException(int count)
         {
-            IList collection = NonGenericIListFactory(count);
-            Assert.All(InvalidValues, value =>
+            if (IsReadOnly && !ExpectedFixedSize)
```
Comment (ianhays) [inline src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs]: should be `if (!IsReadOnly && !ExpectedFixedSize)`

**Comment by ianhays** [comment]
Thanks Hugh, the changes look good minus the one comment!

File: src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs (hunk 1 above)
Comment (hughbe) [inline src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs]: done, thanks Ian!

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please (https://github.com/dotnet/corefx/issues/8926)

**Pull request merged**